### PR TITLE
ArticleInfo: add gadget to repo, with endpoint to fetch JS

### PR DESCRIPTION
**Pull request by MusikAnimal**
The generated file will automatically point to the API for the
current XTools installation. You can pass ?uglify=1 to uglify.

### CODE REVIEW

**Comment by MusikAnimal** [comment]
Doing it this way because ideally we wouldn't make our script to be WMF-specific. For the same reason I don't think we want the uglified version in our repo either. So here there's just an endpoint at `/articleinfo-gadget.js` that will return working JS for the current XTools installation. We want to sync to the wiki, just browse to `/articleinfo-gadget.js?uglify=1` and copy/paste

File: src/AppBundle/Controller/ArticleInfoController.php [hunk 1]
```diff
@@ -98,6 +99,39 @@ public function indexAction(Request $request)
...
+     */
+    public function gadgetAction(Request $request)
+    {
+        $rendered = $this->renderView('articleInfo/articleinfo.js.twig');
+
+        // SUPER hacky, but it works and is safe.
+        if ($request->query->get('uglify') != '') {
+            $rendered = str_replace('$', '\$', trim($rendered));
+            $rendered = str_replace('"', '\"', trim($rendered));
+            $script = "echo \"$rendered\" | tee temp.js >/dev/null && ";
```
Comment (samwilson) [inline src/AppBundle/Controller/ArticleInfoController.php]: This should use a temp file in the sys temp dir, or somewhere. e.g. `$tmpFile = sys_get_temp_dir() . '/xtools_articleinfo_gadget.js';`

File: src/AppBundle/Controller/ArticleInfoController.php [hunk 2]
```diff
@@ -98,6 +99,39 @@ public function indexAction(Request $request)
...
+            $script .= $this->get('kernel')->getRootDir() .
+                '/Resources/node_modules/uglify-es/bin/uglifyjs temp.js --mangle';
+            $script .=  '&& rm temp.js >/dev/null';
+            $process = new Process($script);
+            $process->run();
```
Comment (samwilson) [inline src/AppBundle/Controller/ArticleInfoController.php]: Should throw an exception if the process failed, based on `$process->getErrorOutput()`

**Review by samwilson — CHANGES_REQUESTED** [review]
Looks good! The only other thing I'd say other than these two comments is that the uglified version should contain a full URL to the place that generated the ugly, with commented instructions about updating or something. e.g. "This code was automatically generated and should not be edited here manually. For updates, please copy and paste from x."

**Comment by MusikAnimal** [comment]
@samwilson ready for more review when you've time! All excellent ideas, by the way. I like saying it was automatically generated and not to modify it, because people will :)

**Review by Matthewrbowker — APPROVED** [review]

**Review by samwilson — APPROVED** [review]
Looks absolutely wizard.

**Pull request merged**